### ISSUE TITLE
build issues with postgres 9.2

### ISSUE THREAD

**Issue by woodbri** [pgRouting/pgrouting#396]
```
[develop-trsp-fix] ~/work/pgrouting/build$ cmake -DPOSTGRESQL_BIN=/usr/lib/postgresql/9.2/bin/ ..
-- CMAKE_VERSION: 2.8.12.2
-- PGROUTING_GIT_TAG: pgrouting-2.1.0-rc1
-- PGROUTING_GIT_BUILD: 81
-- PGROUTING_GIT_HASH: 5fa89b2
-- PGROUTING_GIT_BRANCH: develop_2_1_0
-- PGROUTING_VERSION_REVISION_NAME: 5fa89b2 develop_2_1_0
-- PGROUTING_SOURCE_DIR: /home/woodbri/work/pgrouting
-- CMAKE_BINARY_DIR: /home/woodbri/work/pgrouting/build
-- PGROUTING_VERSION_REVISION_HASH: 5fa89b2
-- POSTGRESQL_BIN: /usr/lib/postgresql/9.2/bin/
-- UNIX=1
-- WIN32=
-- POSTGRESQL_PG_CONFIG is /usr/lib/postgresql/9.2/bin/pg_config
-- POSTGRESQL_EXECUTABLE is /usr/lib/postgresql/9.2/bin/postgres
-- POSTGRESQL_VERSION_STRING in FindPostgreSQL.cmake is PostgreSQL 9.2.11
-- POSTGRESQL_INCLUDE_DIR: /usr/include/postgresql/9.4/server
-- POSTGRESQL_LIBRARIES: /usr/lib
-- Boost version: 1.54.0
-- Boost headers were found here: /usr/include
-- Boost version: 1.54.0
-- Found the following Boost libraries:
--   thread
-- Boost version: 1.54.0
-- Found the following Boost libraries:
--   thread
--   system
-- CGAL_INCLUDE_DIR=/usr/include
-- CGAL_LIBRARIES=/usr/lib/libCGAL.so
-- BOOST_THREAD_LIBRARIES=/usr/lib/x86_64-linux-gnu/libboost_thread.so;/usr/lib/x86_64-linux-gnu/libboost_system.so;/usr/lib/x86_64-linux-gnu/libpthread.so
-- GMP_LIBRARIES=/usr/lib/x86_64-linux-gnu/libgmp.so
-- Found CGAL: /usr/include, /usr/lib/libCGAL.so, /usr/lib/x86_64-linux-gnu/libboost_thread.so;/usr/lib/x86_64-linux-gnu/libboost_system.so;/usr/lib/x86_64-linux-gnu/libpthread.so, /usr/lib/x86_64-linux-gnu/libgmp.so
-- POSTGRESQL_VERSION_STRING is PostgreSQL 9.2.11
-- POSTGRESQL_VERSION is 9.2.11
-- PGSQL_VERSION is 92
-- Install directory for libraries is set to /usr/lib/postgresql/9.2/lib
-- Extension directory for SQL files is set to /usr/share/postgresql/9.2/extension
-- Creating lib/pgrouting--2.1.0.sig
-- LIBS_TO_INSTALL=
-- Configuring done
-- Generating done
-- Build files have been written to: /home/woodbri/work/pgrouting/build
[develop-trsp-fix] ~/work/pgrouting/build$ make
Scanning dependencies of target vrppdtw
[  2%] Building C object src/vrppdtw/src/CMakeFiles/vrppdtw.dir/pdp.c.o
/home/woodbri/work/pgrouting/src/vrppdtw/src/pdp.c: In function 'vrppdtw':
/home/woodbri/work/pgrouting/src/vrppdtw/src/pdp.c:455:17: warning: implicit declaration of function 'heap_formtuple' [-Wimplicit-function-declaration]
                 tuple = heap_formtuple(tuple_desc, values, nulls);
                 ^
/home/woodbri/work/pgrouting/src/vrppdtw/src/pdp.c:455:23: warning: assignment makes pointer from integer without a cast [enabled by default]
                 tuple = heap_formtuple(tuple_desc, values, nulls);
                       ^
[  5%] Building CXX object src/vrppdtw/src/CMakeFiles/vrppdtw.dir/pdp_solver.cpp.o
[  5%] Built target vrppdtw
Scanning dependencies of target apsp_johnson
[  7%] Building C object src/apsp_johnson/src/CMakeFiles/apsp_johnson.dir/apsp_johnson.c.o
/home/woodbri/work/pgrouting/src/apsp_johnson/src/apsp_johnson.c: In function 'apsp_johnson':
/home/woodbri/work/pgrouting/src/apsp_johnson/src/apsp_johnson.c:341:5: warning: implicit declaration of function 'heap_formtuple' [-Wimplicit-function-declaration]
     tuple = heap_formtuple(tuple_desc, values, nulls);
     ^
/home/woodbri/work/pgrouting/src/apsp_johnson/src/apsp_johnson.c:341:11: warning: assignment makes pointer from integer without a cast [enabled by default]
     tuple = heap_formtuple(tuple_desc, values, nulls);
           ^
[ 10%] Building CXX object src/apsp_johnson/src/CMakeFiles/apsp_johnson.dir/apsp_johnson_boost_wrapper.cpp.o
/home/woodbri/work/pgrouting/src/apsp_johnson/src/apsp_johnson_boost_wrapper.cpp: In function 'int boost_apsp_johnson(edge_apsp_johnson_t*, int, edge_apsp_johnson_t**, int*, char**)':
/home/woodbri/work/pgrouting/src/apsp_johnson/src/apsp_johnson_boost_wrapper.cpp:100:40: warning: '*((void*)(& ei)+32).__gnu_cxx::__normal_iterator<boost::detail::stored_edge_property<long unsigned int, Edge>*, std::vector<boost::detail::stored_edge_property<long unsigned int, Edge>, std::allocator<boost::detail::stored_edge_property<long unsigned int, Edge> > > >::_M_current' may be used uninitialized in this function [-Wmaybe-uninitialized]
   graph_traits<graph_t>::edge_iterator ei, eend;
                                        ^
[ 10%] Built target apsp_johnson
Scanning dependencies of target apsp_warshall
[ 12%] Building C object src/apsp_warshall/src/CMakeFiles/apsp_warshall.dir/apsp.c.o
/home/woodbri/work/pgrouting/src/apsp_warshall/src/apsp.c: In function 'apsp_warshall':
/home/woodbri/work/pgrouting/src/apsp_warshall/src/apsp.c:419:7: warning: implicit declaration of function 'heap_formtuple' [-Wimplicit-function-declaration]
       tuple = heap_formtuple(tuple_desc, values, nulls);
       ^
/home/woodbri/work/pgrouting/src/apsp_warshall/src/apsp.c:419:13: warning: assignment makes pointer from integer without a cast [enabled by default]
       tuple = heap_formtuple(tuple_desc, values, nulls);
             ^
[ 15%] Building CXX object src/apsp_warshall/src/CMakeFiles/apsp_warshall.dir/apsp_boost_wrapper.cpp.o
[ 15%] Built target apsp_warshall
Scanning dependencies of target astar
[ 17%] Building C object src/astar/src/CMakeFiles/astar.dir/astar.c.o
/home/woodbri/work/pgrouting/src/astar/src/astar.c: In function 'shortest_path_astar':
/home/woodbri/work/pgrouting/src/astar/src/astar.c:510:9: warning: implicit declaration of function 'heap_formtuple' [-Wimplicit-function-declaration]
         tuple = heap_formtuple(tuple_desc, values, nulls);
         ^
/home/woodbri/work/pgrouting/src/astar/src/astar.c:510:15: warning: assignment makes pointer from integer without a cast [enabled by default]
         tuple = heap_formtuple(tuple_desc, values, nulls);
               ^
[ 20%] Building CXX object src/astar/src/CMakeFiles/astar.dir/astar_boost_wrapper.cpp.o
[ 20%] Built target astar
Scanning dependencies of target bd_dijkstra
[ 23%] Building C object src/bd_dijkstra/src/CMakeFiles/bd_dijkstra.dir/bdsp.c.o
/home/woodbri/work/pgrouting/src/bd_dijkstra/src/bdsp.c: In function 'bidir_dijkstra_shortest_path':
/home/woodbri/work/pgrouting/src/bd_dijkstra/src/bdsp.c:436:7: warning: implicit declaration of function 'heap_formtuple' [-Wimplicit-function-declaration]
       tuple = heap_formtuple(tuple_desc, values, nulls);
       ^
/home/woodbri/work/pgrouting/src/bd_dijkstra/src/bdsp.c:436:13: warning: assignment makes pointer from integer without a cast [enabled by default]
       tuple = heap_formtuple(tuple_desc, values, nulls);
             ^
[ 25%] Building CXX object src/bd_dijkstra/src/CMakeFiles/bd_dijkstra.dir/bdsp_core.cpp.o
[ 28%] Building CXX object src/bd_dijkstra/src/CMakeFiles/bd_dijkstra.dir/BiDirDijkstra.cpp.o
[ 28%] Built target bd_dijkstra
Scanning dependencies of target bd_astar
[ 30%] Building C object src/bd_astar/src/CMakeFiles/bd_astar.dir/bdastar.c.o
/home/woodbri/work/pgrouting/src/bd_astar/src/bdastar.c: In function 'bidir_astar_shortest_path':
/home/woodbri/work/pgrouting/src/bd_astar/src/bdastar.c:515:7: warning: implicit declaration of function 'heap_formtuple' [-Wimplicit-function-declaration]
       tuple = heap_formtuple(tuple_desc, values, nulls);
       ^
/home/woodbri/work/pgrouting/src/bd_astar/src/bdastar.c:515:13: warning: assignment makes pointer from integer without a cast [enabled by default]
       tuple = heap_formtuple(tuple_desc, values, nulls);
             ^
[ 33%] Building CXX object src/bd_astar/src/CMakeFiles/bd_astar.dir/MinHeap.cpp.o
[ 35%] Building CXX object src/bd_astar/src/CMakeFiles/bd_astar.dir/bdastar_core.cpp.o
[ 38%] Building CXX object src/bd_astar/src/CMakeFiles/bd_astar.dir/BiDirAStar.cpp.o
[ 38%] Built target bd_astar
Scanning dependencies of target common
[ 41%] Building C object src/common/src/CMakeFiles/common.dir/postgres_connection.c.o
[ 43%] Building CXX object src/common/src/CMakeFiles/common.dir/basePath_SSEC.cpp.o
[ 43%] Built target common
Scanning dependencies of target dijkstra
[ 46%] Building C object src/dijkstra/src/CMakeFiles/dijkstra.dir/dijkstra.c.o
/home/woodbri/work/pgrouting/src/dijkstra/src/dijkstra.c: In function 'shortest_path':
/home/woodbri/work/pgrouting/src/dijkstra/src/dijkstra.c:183:7: warning: implicit declaration of function 'heap_formtuple' [-Wimplicit-function-declaration]
       tuple = heap_formtuple(tuple_desc, values, nulls);
       ^
/home/woodbri/work/pgrouting/src/dijkstra/src/dijkstra.c:183:13: warning: assignment makes pointer from integer without a cast [enabled by default]
       tuple = heap_formtuple(tuple_desc, values, nulls);
             ^
[ 48%] Building C object src/dijkstra/src/CMakeFiles/dijkstra.dir/1_to_many_dijkstra.c.o
/home/woodbri/work/pgrouting/src/dijkstra/src/1_to_many_dijkstra.c: In function 'dijkstra_1_to_many':
/home/woodbri/work/pgrouting/src/dijkstra/src/1_to_many_dijkstra.c:176:7: warning: implicit declaration of function 'heap_formtuple' [-Wimplicit-function-declaration]
       tuple = heap_formtuple(tuple_desc, values, nulls);
       ^
/home/woodbri/work/pgrouting/src/dijkstra/src/1_to_many_dijkstra.c:176:13: warning: assignment makes pointer from integer without a cast [enabled by default]
       tuple = heap_formtuple(tuple_desc, values, nulls);
             ^
[ 51%] Building C object src/dijkstra/src/CMakeFiles/dijkstra.dir/many_to_1_dijkstra.c.o
/home/woodbri/work/pgrouting/src/dijkstra/src/many_to_1_dijkstra.c: In function 'dijkstra_many_to_1':
/home/woodbri/work/pgrouting/src/dijkstra/src/many_to_1_dijkstra.c:165:7: warning: implicit declaration of function 'heap_formtuple' [-Wimplicit-function-declaration]
       tuple = heap_formtuple(tuple_desc, values, nulls);
       ^
/home/woodbri/work/pgrouting/src/dijkstra/src/many_to_1_dijkstra.c:165:13: warning: assignment makes pointer from integer without a cast [enabled by default]
       tuple = heap_formtuple(tuple_desc, values, nulls);
             ^
[ 53%] Building C object src/dijkstra/src/CMakeFiles/dijkstra.dir/many_to_many_dijkstra.c.o
/home/woodbri/work/pgrouting/src/dijkstra/src/many_to_many_dijkstra.c: In function 'dijkstra_many_to_many':
/home/woodbri/work/pgrouting/src/dijkstra/src/many_to_many_dijkstra.c:165:7: warning: implicit declaration of function 'heap_formtuple' [-Wimplicit-function-declaration]
       tuple = heap_formtuple(tuple_desc, values, nulls);
       ^
/home/woodbri/work/pgrouting/src/dijkstra/src/many_to_many_dijkstra.c:165:13: warning: assignment makes pointer from integer without a cast [enabled by default]
       tuple = heap_formtuple(tuple_desc, values, nulls);
             ^
[ 56%] Building CXX object src/dijkstra/src/CMakeFiles/dijkstra.dir/dijkstra_driver.cpp.o
[ 56%] Built target dijkstra
Scanning dependencies of target kdijkstra
[ 58%] Building CXX object src/kdijkstra/src/CMakeFiles/kdijkstra.dir/k_targets_boost_wrapper.cpp.o
[ 61%] Building C object src/kdijkstra/src/CMakeFiles/kdijkstra.dir/k_targets_sp.c.o
/home/woodbri/work/pgrouting/src/kdijkstra/src/k_targets_sp.c: In function 'onetomany_dijkstra_dist':
/home/woodbri/work/pgrouting/src/kdijkstra/src/k_targets_sp.c:621:9: warning: implicit declaration of function 'heap_formtuple' [-Wimplicit-function-declaration]
         tuple = heap_formtuple(tuple_desc, values, nulls);
         ^
/home/woodbri/work/pgrouting/src/kdijkstra/src/k_targets_sp.c:621:15: warning: assignment makes pointer from integer without a cast [enabled by default]
         tuple = heap_formtuple(tuple_desc, values, nulls);
               ^
/home/woodbri/work/pgrouting/src/kdijkstra/src/k_targets_sp.c: In function 'onetomany_dijkstra_path':
/home/woodbri/work/pgrouting/src/kdijkstra/src/k_targets_sp.c:973:15: warning: assignment makes pointer from integer without a cast [enabled by default]
         tuple = heap_formtuple(tuple_desc, values, nulls);
               ^
[ 61%] Built target kdijkstra
Scanning dependencies of target ksp
[ 64%] Building C object src/ksp/src/CMakeFiles/ksp.dir/ksp.c.o
/home/woodbri/work/pgrouting/src/ksp/src/ksp.c: In function 'kshortest_path':
/home/woodbri/work/pgrouting/src/ksp/src/ksp.c:141:7: warning: implicit declaration of function 'heap_form_tuple' [-Wimplicit-function-declaration]
       tuple = heap_form_tuple(tuple_desc, values, nulls);
       ^
/home/woodbri/work/pgrouting/src/ksp/src/ksp.c:141:13: warning: assignment makes pointer from integer without a cast [enabled by default]
       tuple = heap_form_tuple(tuple_desc, values, nulls);
             ^
[ 66%] Building CXX object src/ksp/src/CMakeFiles/ksp.dir/ksp_driver.cpp.o
[ 66%] Built target ksp
Scanning dependencies of target trsp
[ 69%] Building C object src/trsp/src/CMakeFiles/trsp.dir/trsp.c.o
/home/woodbri/work/pgrouting/src/trsp/src/trsp.c: In function 'turn_restrict_shortest_path_vertex':
/home/woodbri/work/pgrouting/src/trsp/src/trsp.c:670:7: warning: implicit declaration of function 'heap_formtuple' [-Wimplicit-function-declaration]
       tuple = heap_formtuple(tuple_desc, values, nulls);
       ^
/home/woodbri/work/pgrouting/src/trsp/src/trsp.c:670:13: warning: assignment makes pointer from integer without a cast [enabled by default]
       tuple = heap_formtuple(tuple_desc, values, nulls);
             ^
/home/woodbri/work/pgrouting/src/trsp/src/trsp.c: In function 'turn_restrict_shortest_path_edge':
/home/woodbri/work/pgrouting/src/trsp/src/trsp.c:819:13: warning: assignment makes pointer from integer without a cast [enabled by default]
       tuple = heap_formtuple(tuple_desc, values, nulls);
             ^
[ 71%] Building CXX object src/trsp/src/CMakeFiles/trsp.dir/trsp_core.cpp.o
[ 74%] Building CXX object src/trsp/src/CMakeFiles/trsp.dir/GraphDefinition.cpp.o
[ 74%] Built target trsp
Scanning dependencies of target tsp
[ 76%] Building C object src/tsp/src/CMakeFiles/tsp.dir/tsp2.c.o
/home/woodbri/work/pgrouting/src/tsp/src/tsp2.c: In function 'tsp_matrix':
/home/woodbri/work/pgrouting/src/tsp/src/tsp2.c:295:9: warning: implicit declaration of function 'heap_formtuple' [-Wimplicit-function-declaration]
         tuple = heap_formtuple(tuple_desc, values, nulls);
         ^
/home/woodbri/work/pgrouting/src/tsp/src/tsp2.c:295:15: warning: assignment makes pointer from integer without a cast [enabled by default]
         tuple = heap_formtuple(tuple_desc, values, nulls);
               ^
[ 79%] Building C object src/tsp/src/CMakeFiles/tsp.dir/tsplib.c.o
[ 79%] Built target tsp
Scanning dependencies of target driving_distance
[ 82%] Building C object src/driving_distance/src/CMakeFiles/driving_distance.dir/alpha.c.o
/home/woodbri/work/pgrouting/src/driving_distance/src/alpha.c: In function 'alphashape':
/home/woodbri/work/pgrouting/src/driving_distance/src/alpha.c:392:7: warning: implicit declaration of function 'heap_formtuple' [-Wimplicit-function-declaration]
       tuple = heap_formtuple(tuple_desc, values, nulls);
       ^
/home/woodbri/work/pgrouting/src/driving_distance/src/alpha.c:392:13: warning: assignment makes pointer from integer without a cast [enabled by default]
       tuple = heap_formtuple(tuple_desc, values, nulls);
             ^
[ 84%] Building CXX object src/driving_distance/src/CMakeFiles/driving_distance.dir/alpha_drivedist.cpp.o
[ 87%] Building CXX object src/driving_distance/src/CMakeFiles/driving_distance.dir/boost_interface_drivedist.cpp.o
[ 89%] Building C object src/driving_distance/src/CMakeFiles/driving_distance.dir/drivedist.c.o
/home/woodbri/work/pgrouting/src/driving_distance/src/drivedist.c: In function 'driving_distance':
/home/woodbri/work/pgrouting/src/driving_distance/src/drivedist.c:165:7: warning: implicit declaration of function 'heap_formtuple' [-Wimplicit-function-declaration]
       tuple = heap_formtuple(tuple_desc, values, nulls);
       ^
/home/woodbri/work/pgrouting/src/driving_distance/src/drivedist.c:165:13: warning: assignment makes pointer from integer without a cast [enabled by default]
       tuple = heap_formtuple(tuple_desc, values, nulls);
             ^
[ 92%] Building C object src/driving_distance/src/CMakeFiles/driving_distance.dir/many_to_dist_driving_distance.c.o
/home/woodbri/work/pgrouting/src/driving_distance/src/many_to_dist_driving_distance.c: In function 'driving_many_to_dist':
/home/woodbri/work/pgrouting/src/driving_distance/src/many_to_dist_driving_distance.c:171:7: warning: implicit declaration of function 'heap_formtuple' [-Wimplicit-function-declaration]
       tuple = heap_formtuple(tuple_desc, values, nulls);
       ^
/home/woodbri/work/pgrouting/src/driving_distance/src/many_to_dist_driving_distance.c:171:13: warning: assignment makes pointer from integer without a cast [enabled by default]
       tuple = heap_formtuple(tuple_desc, values, nulls);
             ^
[ 92%] Built target driving_distance
Scanning dependencies of target vrp_basic
[ 94%] Building C object src/vrp_basic/src/CMakeFiles/vrp_basic.dir/VRP.c.o
/home/woodbri/work/pgrouting/src/vrp_basic/src/VRP.c: In function 'vrp':
/home/woodbri/work/pgrouting/src/vrp_basic/src/VRP.c:892:3: warning: implicit declaration of function 'heap_formtuple' [-Wimplicit-function-declaration]
   tuple = heap_formtuple(tuple_desc, values, nulls);
   ^
/home/woodbri/work/pgrouting/src/vrp_basic/src/VRP.c:892:9: warning: assignment makes pointer from integer without a cast [enabled by default]
   tuple = heap_formtuple(tuple_desc, values, nulls);
         ^
[ 97%] Building CXX object src/vrp_basic/src/CMakeFiles/vrp_basic.dir/VRP_core.cpp.o
[100%] Building CXX object src/vrp_basic/src/CMakeFiles/vrp_basic.dir/VRP_Solver.cpp.o
[100%] Built target vrp_basic
Scanning dependencies of target routing-2.1
Linking CXX shared library lib/librouting-2.1.so
[100%] Built target routing-2.1
[develop-trsp-fix] ~/work/pgrouting/build$ sudo make install
[sudo] password for woodbri:
[  5%] Built target vrppdtw
[ 10%] Built target apsp_johnson
[ 15%] Built target apsp_warshall
[ 20%] Built target astar
[ 28%] Built target bd_dijkstra
[ 38%] Built target bd_astar
[ 43%] Built target common
[ 56%] Built target dijkstra
[ 61%] Built target kdijkstra
[ 66%] Built target ksp
[ 74%] Built target trsp
[ 79%] Built target tsp
[ 92%] Built target driving_distance
[100%] Built target vrp_basic
[100%] Built target routing-2.1
Install the project...
-- Install configuration: ""
-- Installing: /usr/lib/postgresql/9.2/lib/librouting-2.1.so
-- Installing: /usr/share/postgresql/9.2/extension/pgrouting--2.1.0.sql
-- Installing: /usr/share/postgresql/9.2/extension/pgrouting.control
-- Installing: /usr/share/postgresql/9.2/extension/pgrouting--2.0.0--2.1.0.sql
[develop-trsp-fix] ~/work/pgrouting/build$ cd ..
[develop-trsp-fix] ~/work/pgrouting$ tools/test-runner.pl -pguser postgres -pghost localhost -pgport 5432 -clean -alg trsp
RUNNING: test-runner.pl -pguser postgres -pghost localhost -pgport 5432 -clean -alg trsp
CREATE EXTENSION
ERROR:  could not load library "/usr/lib/postgresql/9.2/lib/librouting-2.1.so": /usr/lib/postgresql/9.2/lib/librouting-2.1.so: undefined symbol: palloc
ERROR:  function pgr_version() does not exist
LINE 1: select pgr_version()
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
ERROR: failed to install pgrouting into the database!
```

**Comment by cvvergara** [comment]
https://travis-ci.org/pgRouting/pgrouting/jobs/75988814

**Comment by woodbri** [comment]
This is resolved on master for 2.1. Closing.
